lib/ansible/modules/get_url.py

```diff
@@ -506,12 +507,12 @@ def main():
             for line in lines:
                 parts = line.split(None, 1)
                 if len(parts) == 2:
-                    checksum_map[parts[0]] = parts[1]
+                    checksum_map[sha256(to_bytes(line)).hexdigest()] = (parts[0], parts[1])
```

Comment:
since checksum seems designed to allow for diff hashing algorithms, should this not be configurable depending on which one is used and not just always sha256

Reply:
This is just an internal implementation and doesn't need to be configurable. It is generating a unique hash of the line and using that as the key rather than using the hash itself as the key. The example from the bug report had a file that used the same hash for different files, which resulted in it reporting it could not find the hash.

```
80fcaa0ead2d75283613b9d9d8a02f0df76b7cd018153c2454069cb30ebc0fdb  rhcos-aws.x86_64.vmdk.gz
80fcaa0ead2d75283613b9d9d8a02f0df76b7cd018153c2454069cb30ebc0fdb  rhcos-4.5.6-x86_64-aws.x86_64.vmdk.gz
```

Reply:
As @sivel pointed out, we could just use the entire line as the key.